VirtualListView/VirtualListViewHandler.cs

```diff
@@ -97,36 +97,33 @@ public bool IsItemSelected(int sectionIndex, int itemIndex)
 		return previousSelections.Contains(new ItemPosition(sectionIndex, itemIndex));
 	}
 
-	ItemPosition[] previousSelections = Array.Empty<ItemPosition>();
+	HashSet<ItemPosition> previousSelections = [];
 
 	public static void MapSelectedItems(VirtualListViewHandler handler, IVirtualListView virtualListView)
 	{
 		if (handler is null)
 			return;
 
-		var newSelections = virtualListView?.SelectedItems ?? Array.Empty<ItemPosition>();
+		ISet<ItemPosition> newSelections = virtualListView?.SelectedItems ?? new HashSet<ItemPosition>();
 
 		if (virtualListView.SelectionMode == SelectionMode.None)
```

Comment:
I have a small refactoring suggestion, as i wonder if this needs to be checked for null as well?  I Mean when virtualListView is null, it  throws a NullReferenceException.
So in the current state when we have a virtualListView of null we have a new HashSet on Line 107 but cannot continue, as it should throw an exception on Line 109/111

```suggestion
		if (virtualListView?.SelectionMode == SelectionMode.None)
```